example/lib/main.dart

```diff
@@ -174,6 +174,10 @@ class _MyAppState extends State<MyApp> {
       listenRealtimeConnection(realtime);
       ably.RealtimeChannel channel = realtime.channels.get("test-channel");
       listenRealtimeChannel(channel);
+      setState(() {
+        _realtime = realtime;
+        _realtimeCreationState = OpState.Succeeded;
+      });
     } catch (error) {
```

Comment:
It's not really part of this PR but `realtime` doesn't seem to be used outside `try { ...}`
then there shouldn't be a reason to declare it outside `try { ... }`

I'd change

```dart
    ably.Realtime realtime;
    try {
      realtime = ably.Realtime(options: clientOptions);
```


```dart
    try {
      var realtime = ably.Realtime(options: clientOptions);
```

Reply:
addressed 933e6fddb31504e18fd257121406ff8195c14b06